rdtools/test/analysis_test.py

```diff
@@ -0,0 +1,246 @@
+    # initialize the RdAnalysis class
+    temp = analysis.RdAnalysis(power, power*1000, pvlib_location=loc, 
+                        pv_tilt=meta['tilt'], pv_azimuth=meta['azimuth'], 
+                        temperature_coefficient=0 )
+    # get clearsky expected
+    temp.clearsky_preprocess()  # get expected clear-sky values in clearsky_poa.
+    cs = temp.clearsky_poa
+    
+    rdCS = analysis.RdAnalysis(power*cs, cs, temperature_coefficient=meta['tempco'],
+            pvlib_location=loc, pv_tilt=meta['tilt'], pv_azimuth=meta['azimuth'], 
+            temperature_model = {'a': -3.47, 'b': -0.0594, 'deltaT': 3}
+            )
+    return rdCS
+
+@pytest.fixture()
+def getRdSoiling(normalized_daily):
+    
+    loc = pvlib.location.Location(meta['latitude'], meta['longitude'], tz = meta['timezone'])
+    power = normalized_daily.resample('1h').interpolate()
+    poa = power*0+1000
+    rdSoiling = analysis.RdAnalysis(power,poa, cell_temperature = power*0+25, pvlib_location=loc,
+                        temperature_coefficient=0, interp_freq='D',
+                        pv_tilt=meta['tilt'], pv_azimuth=meta['azimuth'] ) 
+    return rdSoiling
+    
+
+def test_sensor_analysis_fixture(getRd):
+
+    getRd.sensor_analysis(analyses=['yoy_degradation'])
+
+    yoy_results = getRd.results['sensor']['yoy_degradation']
+
+    assert -1 == pytest.approx(yoy_results['p50_rd'], abs=1e-2)
+    assert [-1, -1] == pytest.approx(yoy_results['rd_confidence_interval'], abs=1e-2)
+
+
+def test_clearsky_analysis_fixture(getRdCS):
+
+    getRdCS.clearsky_analysis()
+    cs_yoy_results = getRdCS.results['clearsky']['yoy_degradation']
+
+    assert -4.744 == pytest.approx(cs_yoy_results['p50_rd'], abs=1e-3)
+    assert [-4.75, -4.73] == pytest.approx(cs_yoy_results['rd_confidence_interval'], abs=1e-2)
+    
+    # Re-run while passing some of the clearsky values back into the original instance to improve test coverage
+    poa = getRdCS.poa
+    cspoa = getRdCS.clearsky_poa
+    cscell = getRdCS.clearsky_cell_temperature
+    csamb = getRdCS.clearsky_ambient_temperature
+    pv = getRdCS.pv_energy
+    loc = pvlib.location.Location(meta['latitude'], meta['longitude'], tz = meta['timezone'])
+    
+    rdCS2 = analysis.RdAnalysis(pv, poa=poa, clearsky_cell_temperature=cscell,
+            clearsky_poa=cspoa, clearsky_ambient_temperature=csamb, pv_input='energy', 
+            pvlib_location=loc, pv_tilt=pd.Series(data=meta['tilt'], index=pv.index), 
+            pv_azimuth=pd.Series(data=meta['azimuth'], index=pv.index),
+            temperature_model = {'a': -3.47, 'b': -0.0594, 'deltaT': 3}, interp_freq='1H'
+            )
+    rdCS2.pv_power = getRdCS.pv_power
+    rdCS2.clearsky_analysis()
+    cs_yoy_results2 = rdCS2.results['clearsky']['yoy_degradation']
+    assert -5.1278 == pytest.approx(cs_yoy_results2['p50_rd'], abs=1e-3)
+    assert [-5.1285, -5.1269] == pytest.approx(cs_yoy_results2['rd_confidence_interval'], abs=1e-3)    
```

Comment:
Also confused about this one.  What does this recycling approach do?  (also why is the returned Rd so crazy)

Reply:
Maybe there's a better way to do this, but lines 147 and 148 return the clearsky degradation fixture which is based on passing in power and poa.  That only exercises a couple of the inputs into the clearsky analysis - you can also pass in energy, cs_poa, and cs_temperature.  Lines 154-169 are just increasing test coverage by calculating those values from the first run, and passing them back in.  As for why the calculated value (-5.13) differs from the first run (-4.75), based on ostensibly consistent input values, that's probably something worth looking into.

Reply:
OK, I checked this out, and the two assertion values are the same if you use the same temperature coefficient.  I added that in and changed the test value to match, just to avoid confusion.  Testing a different temperature coefficient could be done in a separate test.

Reply:
Ok thanks for the explanations, I get what's going on now.  The pattern of passing around the object you're testing (rather than arguments to the object) is a new idea to me.  Do you think it makes more sense here than how the [soiling tests](https://github.com/NREL/rdtools/blob/development/rdtools/test/soiling_test.py) work, where all the fixtures are parameters to the function/object?  If it works it works, I'm just thinking about how easy it will be to figure out again when we haven't looked at it for a few months.

Reply:
@cdeline I played around with rewriting these tests in the style I was talking about above.  I'm not convinced I like it better than what you're doing here, but am curious what you think: https://github.com/NREL/rdtools/blob/modelchains_pytests_anderson/rdtools/test/analysis_test.py

Reply:
OK, thanks.  I like your method, and think that it hews more closely to the intent of pytest fixtures, so let's go with that.  I'll bring it into my branch.